src/Microsoft.AspNetCore.Razor.Runtime/Sources/ClosedGenericMatcher.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+// TODO remove this file and use sources packages https://github.com/aspnet/Common/issues/180
+
+namespace Microsoft.Extensions.Internal
+{
+    /// <summary>
+    /// Helper related to generic interface definitions and implementing classes.
+    /// </summary>
+    internal static class ClosedGenericMatcher
```

Comment:
Can we throw all of these shared sources into a more unappealing folder aside from `Sources`? AKa. `__TemporarySources__`. If you can think of a better name go for it. 

`Sources` seems to general and I can envision people adding bits to the folder even when they don't need to because it exists.